knowledge_repo/app/index.py

```diff
@@ -19,8 +24,58 @@
 UNLOCKED = '0'
 
 
+def set_up_indexing_timers(app):
+    if not app.config['INDEXING_ENABLED']:
+        return False
+
+    if app.check_thread_support(check_index=True, check_repositories=app.config['INDEXING_UPDATES_REPOSITORIES']):
+        if os.environ['KNOWLEDGE_REPO_MASTER_UUID'] != app.uuid:
+            logger.info("Not spawning index-sync timers for non-master application instance: {}".format(app.uuid))
+            return
+
+        def index_watchdog(app):
+            while True:
+                if not hasattr(app, 'sync_thread') or not app.sync_thread.is_alive():
+                    logger.warning("Master indexing thread has died. Restarting...")
+                    context = None
```

Comment:
what does this do?

Reply:
Nothing. An entirely vestigial piece of code. Thanks!